### PR TITLE
PR: Add a flag to use case sensitive search/replace

### LINKED ISSUE #4072: Case sensitive replace does not respect case
Performing a Replace ("Search -> Replace Text") with "Case Sensitive" selected does not respect case, i.e. text is matched for any combination of upper and lower case characters and replaced with the "Replace with:" text.

To reproduce:

1.  From the menu, select "File -> New file...".
2. Enter the following text: """
Sample text to replace
"""
3. From the menu, select "Search -> Replace Text"
4. Enter the text "sample" in text to look for
5. Enter the text "Should not appear" in text to replace with
6. Ensure "Case Sensitive" is selected / checked / active.
7. Click "Replace / Find"

Expected:  Text should remain as is because "sample" does not match "Sample" in a case sensitive manner.

Instead: The text "sample" is replaced with the replace text "Should not appear".

## Versions and main components

* Spyder Version: Spyder 3.1.2 
* Python Version: Python 3.6.0 32bits, Qt 5.7.1, PyQt5 5.7.1 on Windows 
* Operating system: Windoes

## Dependencies

jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  None (NOK)
nbconvert >=4.0  :  5.1.1 (OK)
numpy >=1.7      :  None (NOK)
pandas >=0.13.1  :  None (NOK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.6.5 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.5.2 (OK)
sympy >=0.7.3    :  None (NOK)

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #4072

### CODE REVIEW

**Comment by goanpeca** [comment]
Could we add a test :-p ?

**Comment by ccordoba12** [comment]
Yep, please add a test ;-)

File: spyder/utils/fixtures.py [hunk 1]
```diff
@@ -41,3 +43,14 @@ def fin():
...
     request.addfinalizer(fin)
     return CONF
+
+@pytest.fixture
+def editorbot(qtbot):
+    widget = codeeditor.CodeEditor(None)
+    widget.setup_editor(linenumbers=True, markers=True, tab_mode=False,
+                         font=QFont("Courier New", 10),
+                         show_blanks=True, color_scheme='Zenburn')
+    widget.setup_editor(language='Python')
```
Comment (goanpeca) [inline spyder/utils/fixtures.py]: Why is this 2 steps? and not just this (widget.setup_editor(language='Python')) one?

Comment (goanpeca) [inline spyder/utils/fixtures.py]: Are we not using this bot on the tests?

Comment (ccordoba12) [inline spyder/utils/fixtures.py]: Yeah, it seems you added this bot but at the end you're not using it :-/

Comment (dalthviz) [inline spyder/utils/fixtures.py]: Both are right, I forgot to change that bot for the one that I'm using and that is in `test_editor.py` 😖

File: spyder/widgets/tests/test_editor.py [hunk 2]
```diff
@@ -84,6 +60,20 @@ def test_run_last_line_when_nonempty(qtbot):
...
+    text = ' test \nTEST \nTest \ntesT '
+    editor.set_text(text)
+    editorStack.find_widget.search_text.add_text('test')
+    editorStack.find_widget.replace_text.add_text('pass')
+    editorStack.find_widget.replace_find()
+    editorStack.find_widget.replace_find()
+    editorStack.find_widget.replace_find()
+    editorStack.find_widget.replace_find()
+    editor_text = editor.toPlainText()
+    assert editor_text == ' pass \nTEST \nTest \ntesT '
```
Comment (ccordoba12) [inline spyder/widgets/tests/test_editor.py]: Great test!

**Pull request merged**